src/Parser/Core/Modules/Items/Legendaries.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ItemLink from 'common/ItemLink';
+import ITEM_QUALITIES from 'common/ITEM_QUALITIES';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SUGGESTION_IMPORTANCE from 'Parser/Core/ISSUE_IMPORTANCE';
+
+// Make sure to push this once we get to 1000!
+const MAX_LEGENDARY_ILVL = 970;
+
+class Legendaries extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  legendaries = [];
+  debug = false;
+
+  on_finished() {
+    const slots = this.combatants.selected._gearItemsBySlotId;
+    for(let index = 0; index < Object.keys(slots).length; index++) {
+      const item = slots[index];
+      if (item.quality === ITEM_QUALITIES.LEGENDARY) {
+        this.legendaries.push(item);
```

Comment:
We prefer using functional paradigms when possible, so here you should be looping with `slots.forEach( ...`

Reply:
Could also skip the if statement with a `filter`. Check out the functions available to a JS array, there's some pretty good stuff: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array

Reply:
I can't call forEach or filter on slots as the data is formatted like this:
![image](https://user-images.githubusercontent.com/5927637/31568493-f0f53b84-b074-11e7-9fef-60e205683f3d.png)